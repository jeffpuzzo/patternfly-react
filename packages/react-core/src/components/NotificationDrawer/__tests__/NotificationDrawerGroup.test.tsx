import React from 'react';
import { render, screen } from '@testing-library/react';
import { NotificationDrawerGroup } from '../NotificationDrawerGroup';

describe('NotificationDrawerGroup', () => {
  test('renders with PatternFly Core styles', () => {
    render(<NotificationDrawerGroup count={2} isExpanded={false} title="Critical Alerts" />);
    expect(screen.getByText('Critical Alerts').outerHTML).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<NotificationDrawerGroup count={2} isExpanded={false} title="Critical Alerts" className="extra-class" />);
    expect(screen.getByText('Critical Alerts').className).toMatchSnapshot();
  });

  test('drawer group with isExpanded applied ', () => {
    render(<NotificationDrawerGroup count={2} isExpanded title="Critical Alerts" />);
    expect(screen.getByText('Critical Alerts').outerHTML).toMatchSnapshot();
  });

  test('drawer group with isRead applied ', () => {
    render(<NotificationDrawerGroup count={2} isExpanded={false} isRead={true} title="Critical Alerts" />);
    expect(screen.getByText('Critical Alerts').outerHTML).toMatchSnapshot();
  });
});
