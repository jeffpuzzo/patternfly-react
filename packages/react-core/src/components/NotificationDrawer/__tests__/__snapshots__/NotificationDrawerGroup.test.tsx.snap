// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawerGroup className is added to the root element 1`] = `"pf-c-notification-drawer__group-toggle-title"`;

exports[`NotificationDrawerGroup drawer group with isExpanded applied  1`] = `"<div class=\\"pf-c-notification-drawer__group-toggle-title\\">Critical Alerts</div>"`;

exports[`NotificationDrawerGroup drawer group with isRead applied  1`] = `"<div class=\\"pf-c-notification-drawer__group-toggle-title\\">Critical Alerts</div>"`;

exports[`NotificationDrawerGroup renders with PatternFly Core styles 1`] = `"<div class=\\"pf-c-notification-drawer__group-toggle-title\\">Critical Alerts</div>"`;
